R/tmb-methods.R

```diff
@@ -192,34 +194,163 @@ h_get_prediction_variance <- function(object, n_sim, tmb_data) {
 #' model.frame(object)
 #' model.frame(object, include = "subject_var")
 model.frame.mmrm_tmb <- function(formula, data, include = NULL, full, na.action = "na.omit", ...) { # nolint
-  include_choice <- c("subject_var", "visit_var", "group_var", "response_var")
+  # construct updated formula and data arguments
+  lst_formula_and_data <-
+    h_construct_model_frame_inputs(
+      formula = formula,
+      data = data,
+      include = include,
+      full = full
+    )
+
+  # construct data frame to return
+  ret <-
+    stats::model.frame(
+      formula = lst_formula_and_data$formula,
+      data = lst_formula_and_data$data,
+      na.action = na.action
+    )
+  # we need the full formula obs, so recalculating if not already full
+  ret_full <-
+    switch(as.character(lst_formula_and_data$is_full),
+      "TRUE" = ret,
+      "FALSE" =
+        stats::model.frame(
+          formula = lst_formula_and_data$formula_full,
+          data = lst_formula_and_data$data
+        )
+    )
+
+  # subset data frame to only include obs included in model
+  ret[rownames(ret) %in% rownames(ret_full), , drop = FALSE]
+}
+
+
+#' Construct `model.frame()` formula and data inputs from mmrm model fit
+#'
+#' @description
+#' Function returns a named list with three elements:
+#' - `"formula"`: the formula including the columns requested in the `include=` argument.
+#' - `"formula_full"`: the formula including all columns
+#' - `"data"`: a data frame including all columns where factor and
+#'   character columns have been processed with `h_factor_ref()`.
```

Comment:
```suggestion
#'   character columns have been processed with [h_factor_ref()].
```
and similar below to automatically create links to the doc pages